File: RedSheet.DbEntities/Models/vRelationshipRequire.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vRelationshipRequires", Schema = "dbo")]
	[Serializable]
	public partial class vRelationshipRequire  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int RelationshipRequireId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string RelationshipRequireName { get; set; } 

	public vRelationshipRequire()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-relationship-require.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vRelationshipRequire {
    constructor(vRelationshipRequire?: vRelationshipRequire )  {
        let properties = [ "className", "relationshipRequireId", "relationshipRequireName",];
        for (let property of properties)
            if (vRelationshipRequire && vRelationshipRequire[property])
                this[property] = vRelationshipRequire[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	relationshipRequireId : number =   0 ;
 
    @required()
    @maxLength(200)
	relationshipRequireName : string =   undefined;


}
